cime/src/drivers/mct/main/cime_comp_mod.F90

```diff
@@ -3741,24 +3807,27 @@ subroutine cime_run_glc_setup_send(lnd2glc_averaged_now)
        call t_drvstartf ('CPL:GLCPREP',cplrun=.true.,barrier=mpicom_CPLID)
        if (drv_threading) call seq_comm_setnthreads(nthreads_CPLID)
 
-       if (lnd_c2_glc) then
-          ! NOTE - only create appropriate input to glc if the avg_alarm is on
-          if (glcrun_avg_alarm) then
+       ! NOTE - only create appropriate input to glc if the avg_alarm is on
+       if (glcrun_avg_alarm) then
+          if (lnd_c2_glc .or. ocn_c2_glc) then
              call prep_glc_accum_avg(timer='CPL:glcprep_avg')
-             lnd2glc_averaged_now = .true.
 
-             ! Note that l2x_gx is obtained from mapping the module variable l2gacc_lx
-             call prep_glc_calc_l2x_gx(fractions_lx, timer='CPL:glcprep_lnd2glc')
+             if (lnd_c2_glc) then
+                lnd2glc_averaged_now = .true.
+                ! Note that l2x_gx is obtained from mapping the module variable l2gacc_lx
+                call prep_glc_calc_l2x_gx(fractions_lx, timer='CPL:glcprep_lnd2glc')
 
-             call prep_glc_mrg(infodata, fractions_gx, timer_mrg='CPL:glcprep_mrgx2g')
+                call prep_glc_mrg_lnd(infodata, fractions_gx, timer_mrg='CPL:glcprep_mrgx2g')
+             endif
 
              call component_diag(infodata, glc, flow='x2c', comment='send glc', &
                   info_debug=info_debug, timer_diag='CPL:glcprep_diagav')
 
-          else
-             call prep_glc_zero_fields()
-          end if  ! glcrun_avg_alarm
-       end if  ! lnd_c2_glc
+          endif ! lnd_c2_glc or ocn_c2_glc
+
+       else
+          call prep_glc_zero_fields()
```

Comment:
Previously this call to `prep_glc_zero_fields` was only done if `lnd_c2_glc` was true. Now it's being done regardless of the value of `lnd_c2_glc` or `ocn_c2_glc`. I'm wondering if it should only be done if `lnd_c2_glc .or. ocn_c2_glc`. It may be okay as is, but can you confirm?

Reply:
@billsacks , funny, I had just been thinking last night that this was inadvertently changed and I need to take a closer look at it.  I'll admit I don't fully understand the requirements here, but the safest thing would be to retain the previous functionality as written, so I've added a commit that does that (312aa4fb348f8b5584ae612a4692a81860d5eb65).